app/models/scheme.rb

```diff
@@ -5,7 +5,8 @@ class Scheme < ApplicationRecord
 
   scope :search_by_code, ->(code) { where("code ILIKE ?", "%#{code}%") }
   scope :search_by_service_name, ->(name) { where("service_name ILIKE ?", "%#{name}%") }
-  scope :search_by, ->(param) { search_by_code(param).or(search_by_service_name(param)) }
+  scope :search_by_postcode, ->(postcode) { joins(:locations).where("locations.postcode ILIKE ?", "%#{postcode.delete(' ')}%") }
```

Comment:
Is the `.delete(' ')` still needed here?

Reply:
I think so, the user can still search with a space